docs/developer/guide.md

```diff
@@ -19,3 +19,40 @@ The following steps are needed to introduce a new resource and its respective re
 - Reference the new resource in [pkg/options/resource.go](https://github.com/kubernetes/kube-state-metrics/blob/master/pkg/options/resource.go).
 - Add a sample Kubernetes manifest to be used by tests in the [tests/manifests/](https://github.com/kubernetes/kube-state-metrics/tree/master/tests/manifests) directory.
 - Lastly, and most importantly, actually implement your new resource(s) and its test binary in [internal/store](https://github.com/kubernetes/kube-state-metrics/tree/master/internal/store). Follow the formatting and structure of other resources.
+
+### Squash Commit History
+
+If your PR still has multiple commits after amending previous commits, you must squash multiple commits into a single commit before your PR can be merged. You can check the number of commits on your PR’s Commits tab or by running git log locally. Squashing commits is a form of rebasing.
```

Comment:
I'm not a fan of this as dogmatic advice, I think encouraging logically splitting commits and squashing "addressed review" type commits makes perfect sense but always squashing is not universally good I believe.